patches/server/0833-Friction-API.patch

```diff
@@ -0,0 +1,153 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Wed, 15 Sep 2021 20:44:22 +0200
+Subject: [PATCH] Friction API
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index fe7e22d9a69d69dfcce63faa28e90945ea45fc49..8683161b8850cd5f7c806cba178b7792c109b5c9 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -260,6 +260,7 @@ public abstract class LivingEntity extends Entity {
+     public boolean bukkitPickUpLoot;
+     public org.bukkit.craftbukkit.entity.CraftLivingEntity getBukkitLivingEntity() { return (org.bukkit.craftbukkit.entity.CraftLivingEntity) super.getBukkitEntity(); } // Paper
+     public boolean silentDeath = false; // Paper - mark entity as dying silently for cancellable death event
++    public io.papermc.paper.entity.Frictionable.FrictionState frictionState = io.papermc.paper.entity.Frictionable.FrictionState.UNSET; // Paper
+ 
+     @Override
+     public float getBukkitYaw() {
+@@ -702,7 +703,7 @@ public abstract class LivingEntity extends Entity {
+     }
+ 
+     public boolean shouldDiscardFriction() {
+-        return this.discardFriction;
++        return this.frictionState == io.papermc.paper.entity.Frictionable.FrictionState.IGNORE || (this.discardFriction && this.frictionState != io.papermc.paper.entity.Frictionable.FrictionState.ALWAYS); // Paper
+     }
+ 
+     public void setDiscardFriction(boolean noDrag) {
+@@ -726,6 +727,11 @@ public abstract class LivingEntity extends Entity {
+ 
+     @Override
+     public void addAdditionalSaveData(CompoundTag nbt) {
++        // Paper start
++        if (this.frictionState != io.papermc.paper.entity.Frictionable.FrictionState.UNSET) {
++            nbt.putString("paper.FrictionState", this.frictionState.toString());
```

Comment:
Paper stores additionally NBT data with the `Paper` prefix, not `paper` (see other examples in `net.minecraft.world.entity.Entity#saveWithoutId(CompoundTag)`).

This applies to obviously any of these references :+1: